miden-lib/asm/miden/kernels/tx/tx.masm

```diff
@@ -85,71 +89,180 @@ proc.increment_num_created_notes
-
+export.create_note   
     # validate the note type
     # NOTE: encrypted notes are currently unsupported `dup.6 push.ENCRYPTED_NOTE eq or`
-    dup.5 push.OFFCHAIN_NOTE eq dup.6 push.PUBLIC_NOTE eq or assert.err=ERR_INVALID_NOTE_TYPE
-    # => [ASSET, tag, note_type, RECIPIENT]
+    dup.2 push.OFFCHAIN_NOTE eq dup.3 push.PUBLIC_NOTE eq or assert.err=ERR_INVALID_NOTE_TYPE
+    # => [tag, aux, note_type, RECIPIENT]
 
     # validate the tag against the note type
-    dup.5 dup.5 u32split
-    # => [tag_high, tag_low, note_type, ASSET, tag, note_type, RECIPIENT]
+    dup.2 dup.1 u32split
+    # => [tag_high, tag_low, note_type, tag, aux, note_type, RECIPIENT]
 
     assertz.err=ERR_NOTE_INVALID_TAG_HIGH_BIT_SET
-    # => [tag_low, note_type, ASSET, tag, note_type, RECIPIENT]
+    # => [tag_low, note_type, tag, note_type, RECIPIENT]
 
     u32shr.30 u32and assertz.err=ERR_NOTE_INVALID_TAG_PREFIX_FOR_TYPE
-    # => [ASSET, tag, note_type, RECIPIENT]
+    # => [tag, aux, note_type, RECIPIENT]
 
     # get the index for the next note to be created and increment counter
     exec.increment_num_created_notes
-    # => [note_idx, ASSET, tag, note_type, RECIPIENT]
+    # => [note_idx, tag, aux, note_type, RECIPIENT]
 
     # get a pointer to the memory address at which the note will be stored
     exec.memory::get_created_note_ptr
-    # => [note_ptr, ASSET, tag, note_type, RECIPIENT]
+    # => [note_ptr, tag, aux, note_type, RECIPIENT]
 
     # populate the metadata
-    movup.5 exec.account::get_id movup.7
-    # => [note_type, sender_acct_id, tag, note_ptr, ASSET, RECIPIENT]
-
-    push.0 # TODO: allow the user to push the aux data
-    # => [aux, note_type, sender_acct_id, tag, note_ptr, ASSET, RECIPIENT]
-
+    movdn.3 movdn.2 exec.account::get_id movdn.2
+    # => [aux, note_type, sender_acct_id, tag, note_ptr, RECIPIENT]
+    
     emit.NEW_NOTE_EVENT
 
     # set the metadata for the new created note
     dup.4 exec.memory::set_created_note_metadata
-    # => [note_ptr, ASSET, RECIPIENT]
+    # => [note_ptr, RECIPIENT]
 
-    # set the number of assets for the note to 1
-    push.1 dup.1 exec.memory::set_created_note_num_assets
-    # => [note_ptr, ASSET, RECIPIENT]
+    # set the RECIPIENT
+    dup movdn.5 exec.memory::set_created_note_recipient
+    # => [note_ptr]
 
-    movdn.4 padw swapw movup.8
-    # => [note_ptr, ASSET, 0, 0, 0, 0, RECIPIENT]
+    # set the number of assets to 0
+    dup push.0 swap exec.memory::set_created_note_num_assets
+    # => [note_ptr]
+end
 
-    # add the asset to the note
-    dup movdn.5 exec.memory::get_created_note_asset_data_ptr mem_storew dropw
-    # => [note_ptr, 0, 0, 0, 0, RECIPIENT]
+#! Adds an ASSET to the specified note and returns a pointer to the memory address
+#! at which the note is stored.
+#!
+#! Inputs: [ptr, ASSET]
+#! Outputs: [ptr]
+#!
+#! ptr is the pointer to the memory address at which the note is stored.
+#! ASSET can be a fungible or non-fungible asset.
+export.move_asset_to_note
```

Comment:
Now seeing how this works, I think `add_asset_to_note` would have been the correct name.